### ISSUE TITLE
additional pixel perfect SCALED display flag

### ISSUE THREAD

**Issue by SigilQuiver** [pygame/pygame#2611]
Currently, the SCALED display flag works differently when the display is in fullscreen/maximized, and when the display is windowed. If the RESIZED flag is enabled, the display uses pixel-perfect scaling unless the window is maximized. If the display is maximized or in fullscreen, the display surface stretches the image to fit the smallest dimension of the screen.

It would be useful if there were two SCALED flags - one for a stretch scale and another for pixel perfect scale.

**Comment by galexandreg** [comment]
I think it's a good idea. I looked at it quickly and it seems possible to override the "pixel-perfect" setting in windowed mode by using the SDL_HINT_RENDER_SCALE_QUALITY environment variable, but fullscreen always sets SDL_RenderSetIntegerScale to false to "stretch" the image. 🤷‍♂️

#1365

**Comment by robertpfeiffer** [comment]
Yeah. The behaviour is intentional (maximising screen use when maximised), but I see no harm in making this configurable, as long as the current defaults stay in place.

**Comment by FinFetChannel** [comment]
> I think it's a good idea. I looked at it quickly and it seems possible to override the "pixel-perfect" setting in windowed mode by using the SDL_HINT_RENDER_SCALE_QUALITY environment variable, but fullscreen always sets SDL_RenderSetIntegerScale to false to "stretch" the image. 🤷‍♂️
> 
> #1365

How do I set this SDL_HINT_RENDER_SCALE_QUALITY  variable to false from Pygame?

**Comment by FinFetChannel** [comment]
Would be nice to have more flexible options, maybe like SCALE_NEAREST and SCALE_CUBIC, as it is, the SCALE flag can be a bit frustrating sometimes, and there is no specification of what exactly it does.
